core/src/main/scala/aima/core/search/adversarial/MinimaxSearch.scala

```diff
@@ -0,0 +1,44 @@
+package aima.core.search.adversarial
+
+import aima.core.search.adversarial.{Game, UtilityValue}
+import aima.core.agent.{Action, NoAction}
+
+/**
+  * @author Aditya Lahiri
+  * @author Shawn Garner
+  */
+
+object MinimaxDecision {
+    def minMaxDecision[S, A, P](g: Game[S][A][P]): S, P => A = {
```

Comment:
I'm not a big fan of single letter parametric names when they actually mean things.  I'd use upper case and use the full name like `STATE`, `ACTION`, and `PERCEPT` rather than `S`, `A`, and `P` respectively.